integration/query_projection_test.go

```diff
@@ -252,3 +252,35 @@ func TestQueryProjectionErrors(t *testing.T) {
 		})
 	}
 }
+
+func TestQueryProjectionSuccess(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setup.Setup(t, shareddata.Scalars, shareddata.Composites)
+	for name, tc := range map[string]struct { //nolint:vet // used for testing only
+		filter     bson.D // required
+		projection any    // required
+
+		skip       string              // optional, skip test with a specified reason
+	}{
+		"QueryProjection_v": {
+			filter:     bson.D{},
+			projection: bson.D{{"v", true}, {"_id", false}},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			if tc.skip != "" {
+				t.Skip(tc.skip)
+			}
+
+			t.Parallel()
+
+			require.NotNil(t, tc.filter, "filter should be set")
+			require.NotNil(t, tc.projection, "projection should be set")
+
+			res, err := collection.Find(ctx, tc.filter, options.Find().SetProjection(tc.projection))
```

Comment:
Let's check `res` against expected value :hugs:

For the `QueryProjection_v` test case I suggest using filter such as `bson.D{{"_id", "int32"}}` so we can assert that expected value of a single document, that'll be `bson.D{{"v", int32(42)}}` for that filter.
FYI, document with `_id` of `int32` exists `shareddata.Scalars` collection,  but not in `shareddata.Composites` so we can remove it for this test:hugs:

Reply:
> Let's check `res` against expected value 🤗
> 
> For the `QueryProjection_v` test case I suggest using filter such as `bson.D{{"_id", "int32"}}` so we can assert that expected value of a single document, that'll be `bson.D{{"v", int32(42)}}` for that filter. FYI, document with `_id` of `int32` exists `shareddata.Scalars` collection, but not in `shareddata.Composites` so we can remove it for this test🤗

I thought of checking the res against expected value but the filter i was using was `bson.D{{"v", 42}}` due to which i was getting an array of results. Hence i thought of checking only the validity of projection by checking if the error is nil.

But i have a query. Should we depend upon the value of response ? Because the test case is used to check if Query Projection is working fine. Even a slightest change in response will lead to test case failure even though the projection works fine which is not what this test case reflects.

Reply:
Yes we should check the response. `Scalar` collection is good one to use because it is frozen, so it won't be modified, see https://github.com/FerretDB/FerretDB/blob/main/integration/shareddata/scalars.go#L39